packages/superset-ui-dimension/src/getTextDimension.ts

```diff
@@ -0,0 +1,54 @@
+import { TextStyle } from './types';
+
+const SVG_NS = 'http://www.w3.org/2000/svg';
+const STYLE_FIELDS: (keyof TextStyle)[] = [
+  'font',
+  'fontWeight',
+  'fontStyle',
+  'fontSize',
+  'fontFamily',
+  'letterSpacing',
+];
+
+export interface GetTextDimensionInput {
+  className?: string;
+  container?: HTMLElement;
+  style?: TextStyle;
+  text: string;
+}
+
+const DEFAULT_DIMENSION = { height: 20, width: 100 };
+
+export default function getTextDimension(
+  input: GetTextDimensionInput,
+  defaultDimension = DEFAULT_DIMENSION,
+) {
+  const { text, className, style = {}, container = document.body } = input;
+
+  const textNode = document.createElementNS(SVG_NS, 'text');
+  textNode.textContent = text;
+
+  if (className !== undefined) {
```

Comment:
I think `className?: string` allows `className` to be both `null` and `undefined`. Do we want to handle both cases with `className != null` here?

Reply:
done